locallib.php

```diff
@@ -965,72 +966,99 @@ public function distrubute_choices() {
     public function synchronize_allocation_and_grouping() {
         require_capability('moodle/course:managegroups', $this->context);
 
-        $groupingidname = ratingallocate_MOD_NAME . '_instid_' . $this->ratingallocateid;
         // Search if there is already a grouping from us.
-        $grouping = groups_get_grouping_by_idnumber($this->course->id, $groupingidname);
-        $groupingid = null;
-        if (!$grouping) {
+        if (!$this->db->record_exists(this_db\ratingallocate_id_grouping::TABLE, ['ratingallocateid' => $this->ratingallocateid])) {
```

Comment:
If you have an if-else statement were you want to check whether a record exists and continue working with the existing record, it is better to check by using `if(!$obj=get_record(...))` (see line 155 in lib.php). This way you only have to call the DB once for this and can continue with $obj for the if or else statement.